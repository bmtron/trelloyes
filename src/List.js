import React, { Component } from 'react';
import Card from './Card';
import './List.css';

export default function List(props){
    return (
      <section className='List'>
        <header className='List-cards'>
          {props.header}
        </header>
        <div className='List-cards'>
          {props.cards.map((card, index) => {
            return <Card onDeleteCard={props.onDeleteCard} id={card.id} listId={props.id} key={index} title={card.title} content={card.content} />;
          })}
          <button onClick={() => props.onRandomCard(props.id)} type='submit'>Add Random Card</button>
        </div>
        
      </section>
    )
}