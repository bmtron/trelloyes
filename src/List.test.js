import React from 'react';
import List from './List';
import Card from './Card'
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';

describe('List component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List cards={[]}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    
    it('renders the UI as expected', () => {
        const tree = renderer
        .create(<List cards={[{id: 4, title: 'Ace', content: 'Strang'}, {id: 3, title: 'King', content: 'Strangggg'}]}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
