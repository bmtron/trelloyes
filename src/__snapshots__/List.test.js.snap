// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-cards"
  />
  <div
    className="List-cards"
  />
</section>
`;

exports[`renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-cards"
  />
  <div
    className="List-cards"
  />
</section>
`;
